design/read-only-registry.md

```diff
@@ -0,0 +1,79 @@
+# Support for Read-Only Registries
+
+## Goals
+
+- Configure kotsadm to use application images from a private registry
+- Support using read-only registry during initial installation
+- Support using read-only registry on application updates
+- Support using read-only registry when settings are modified on the Registry Settings page in Admin Console
+- Support readon-only registries in airgap and on-line modes
+
+## Non Goals
+
+- Delivering images to the read-only registry.
+- Changing registry settings in airgap mode
+
+## Background
+
+Some production enviroments restrict image registry use to read-only mode.  This is an environment wide restriction.  Images are side-loaded into the registry from a trusted internal source.  Right now kotsadm will always push images to the private registry, making it impossible to use in such an environment.
+
+## Detailed Design
+
+### Install
+
+A new argument will be added to `kots install` command to flag the registry as a read-only registry.
+
+```
+kubectl kots install --kotsadm-registry my.private.registry/myapp \
+  --registry-username pulluser \
+  --registry-password pullpassword \
+  --registry-is-readonly \
```

Comment:
The semantics feel a bit off to me. This isn't **just** about having the registry be read-only, but also then skipping pushing any images to the registry, and requiring some out-of-band way to side-load images into the registry. I don't have a suggestion for improvement, but I can imagine some folks getting confused by this.